src/Elastic.Markdown/Myst/Renderers/LlmMarkdown/LlmBlockRenderers.cs

```diff
@@ -39,19 +39,38 @@ public static void RenderBlockWithIndentation(LlmMarkdownRenderer renderer, Mark
 	}
 
 	/// <summary>
-	/// Converts relative URLs to absolute URLs using BuildContext.CanonicalBaseUrl for better LLM consumption
+	/// Converts relative URLs to absolute URLs using BuildContext.CanonicalBaseUrl for better LLM consumption.
+	/// For documentation URLs, converts them to .md format for consistency with LLM-generated files.
 	/// </summary>
 	public static string? MakeAbsoluteUrl(LlmMarkdownRenderer renderer, string? url)
+	{
+		if (renderer.BuildContext.CanonicalBaseUrl == null)
+			return url;
+
+		// For documentation URLs (starting with /docs/), convert to .md format for LLM consistency
+		if (!string.IsNullOrEmpty(url) && url.StartsWith("/docs/", StringComparison.Ordinal))
```

Comment:
I'm not sure if this is the best way to identify an internal URL.

Let me do some digging on this.